src/Domain/LeanCode.CQRS.Execution/QueryLoggerElement.cs

```diff
@@ -0,0 +1,22 @@
+
+using LeanCode.Contracts;
+using LeanCode.Contracts.Security;
+using LeanCode.Pipelines;
+
+namespace LeanCode.CQRS.Execution
+{
+    public class QueryLoggerElement<AppContext, TObj, TRes> : IPipelineElement<AppContext, TObj, TRes>
+        where AppContext : notnull, IPipelineContext
+    {
+        public readonly Serilog.ILogger logger = Serilog.Log.ForContext<QueryLoggerElement<AppContext, TObj, TRes>>();
+
+        public Task<TRes> ExecuteAsync(
+            AppContext ctx,
+            TObj input,
+            Func<AppContext, TObj, Task<TRes>> next)
+        {
+            logger.Information("Query executed with response {@Response}", input);
```

Comment:
If you don't limit the `TObj` to be a `IQuery`, then this can be used for objects different than queries (e.g. operations, or even commands).

This is a very good approach, but let's not use `query` in names in that case.